lib/api/pool.go

```diff
@@ -26,16 +26,36 @@ type PoolMetadata struct {
 }
 
 type PoolSpec struct {
-	IPIP        bool `json:"ipip,omitempty"`
-	NATOutgoing bool `json:"nat-outgoing"`
-	Ipam        bool `json:"ipam"`
-	Disabled    bool `json:"disabled"`
+	// Contains configuration for ipip tunneling
+	// for this pool. If not specified, then ipip
+	// tunneling is disabled for this pool.
+	IPIP *IPIPConfiguration `json:"ipip,omitempty"`
+
+	// When nat-outgoing is true, packets sent from Calico networked
+	// containers in this pool to destinations outside of this pool
+	// will be masqueraded.
+	NATOutgoing bool `json:"nat-outgoing,omitempty"`
+
+	// When disabled is true, Calico IPAM will not assign
+	// addreses from this pool.
+	Disabled bool `json:"disabled,omitempty"`
+}
+
+type IPIPConfiguration struct {
```

Comment:
In case we need additional configuration for `ipip` in the future.

For example if we want to expose the interface name.  For now that defaults to `tunl0`, but maybe one day we'll want to make it configurable on this API.